bin/liquidator/src/rebalance.rs

```diff
@@ -266,6 +290,45 @@ impl Rebalancer {
         Ok((in_token.mint, in_amount))
     }
 
+    fn can_use_sanctum_for_token(&self, mint: Pubkey) -> anyhow::Result<bool> {
+        if !self.config.use_sanctum {
+            return Ok(false);
+        }
+
+        let token = self.mango_client.context.token_by_mint(&mint)?;
+
+        let is_lst = self.is_lst(mint);
+
+        // forbid swapping to something that could be used in another sanctum swap, creating a cycle
+        let is_an_alt_for_sanctum = self
+            .config
+            .alternate_sanctum_route_tokens
+            .contains(&token.token_index);
+
+        Ok(is_lst && !is_an_alt_for_sanctum)
+    }
+
+    async fn get_jupiter_alt_route(
+        &self,
+        quote_fetcher: impl Fn(&u16) -> anyhow::Result<(Pubkey, Pubkey, u64, bool)>,
+        output_mint: Pubkey,
+        jupiter_version: swap::Version,
+        quote_mint: Pubkey,
+    ) -> anyhow::Result<(TransactionBuilder, swap::Quote)> {
+        let mut alt_jobs = vec![];
+        for in_token_index in &self.config.alternate_jupiter_route_tokens {
+            let args = quote_fetcher(in_token_index)?;
+            alt_jobs.push(self.swap_quote(args.0, args.1, args.2, args.3, jupiter_version));
```

Comment:
hmm, I wonder how this could become easier to read.

maybe: `let (in, out, amount, direct) = quote_fetcher(...)?`

Or, maybe preferably, the self.swap_quote() call could be part of the quote_fetcher function?